### PR TITLE
Feature/eventlog ip address with proxy

### LINKED ISSUE #5073: Use original user's IPAddress in logging and IPFilter when behind (e.g.) Cloudflare
## Description of problem
Is your enhancement request related to a problem? Provide a clear and concise description of the problem.
When a site is behind (a service like) Cloudflare, the user's original IPAddress is not the host address from the request, but instead it's passed in a Request Header. DNN's Login page is already able to handle that and log the oringial user's IPAddress by using the UserRequestIPAddressController class.
However, this method of correctly determining the IP Address isn't applied in other places, like logging 404's and for the IPFiltering.

## Description of solution
UserRequestIPAddressController class should be used wherever a user's IPAddress is requested.
Preferably, there should also be a (host level)  setting in the Personabar to set the "UserRequestIPHeader" setting.

## Description of alternatives considered
Given how services like cloudflare work, there really aren't any, afaik.

## Affected browser
* [x] Chrome
* [x] Firefox
* [x] Safari
* [x] Internet Explorer
* [x] Edge
(all, as this isn't browser related)

## Affected DNN Versions
* [x] 9.10.2

### PR DESCRIPTION
**Pull request by skamphuis**
Closes #5073

## Summary
Changed the places where DNN didn't determine IPAddress by using the UserRequestIPAddressController class yet, to use that.
I found these occurences in:
* DNN Platform/Library/Common/Utilities/NetworkUtils.cs
* DNN Platform/Library/Entities/Modules/UserModuleBase.cs
* DNN Platform/Library/Entities/Urls/UrlRewriterUtils.cs
* DNN Platform/Library/Services/Exceptions/SecurityException.cs

### CODE REVIEW

**Review by valadas — APPROVED** [review]
Nice one, I did know even know about that class!

**Comment by skamphuis** [comment]
> Nice one, I did know even know about that class!

tbh: me neither. Ran into the issue after putting a client behind Cloudflare and we got pointed out this issue. When trying to figure out how to fix that, I found that it was implemented already, for a large part.

File: DNN Platform/Library/Services/Exceptions/SecurityException.cs [hunk 1]
```diff
@@ -41,69 +42,72 @@ public SecurityException(string message)
...
+            }
+        }
+
+        private void InitilizePrivateVariables()
+        {
+            // Try and get the Portal settings from httpcontext
+            try
+            {
+                var userRequestIpAddressController = UserRequestIPAddressController.Instance;
+                var ipAddress = userRequestIpAddressController.GetUserRequestIPAddress(new HttpRequestWrapper(request));
```
Comment (bdukes) [inline DNN Platform/Library/Services/Exceptions/SecurityException.cs:88]:
```suggestion
                var ipAddress = userRequestIpAddressController.GetUserRequestIPAddress(new HttpRequestWrapper(HttpContext.Current.Request));
```

Or, alternatively,
```suggestion
                var ipAddress = userRequestIpAddressController.GetUserRequestIPAddress(new HttpRequestWrapper(HttpContextSource.Current.Request));
```

**Comment by skamphuis** [comment]
Thanks for your improvements, @bdukes 
I just committed it.

**Review by mitchelsellers — APPROVED** [review]

File: DNN Platform/Library/Services/Exceptions/SecurityException.cs [hunk 2]
```diff
@@ -41,69 +42,72 @@ public SecurityException(string message)
...
+        {
+            // Try and get the Portal settings from httpcontext
+            try
+            {
+                var userRequestIpAddressController = UserRequestIPAddressController.Instance;
+                var ipAddress = userRequestIpAddressController.GetUserRequestIPAddress(new HttpRequestWrapper(HttpContext.Current.Request));
+
+                if (ipAddress != null)
+                {
+                    this.m_IP = HttpContext.Current.Request.UserHostAddress;
```
Comment (bdukes) [inline DNN Platform/Library/Services/Exceptions/SecurityException.cs:92]:
```suggestion
                    this.m_IP = ipAddress;
```

**Review by bdukes — APPROVED** [review]

**Pull request merged**